### DISCUSSION TITLE
Can we register routes with handler functions instead of file based routing?

### DISCUSSION THREAD

**Discussion by francisashley** [nitrojs/nitro#2042] [Q&A]
Last I checked (March last year) this wasn't possible. It's really frustrating to work with. Ideally all routes were registered in a single routes.ts file and customer handles passed in directly. Has this feature been added in a different way or are there prospects of adding it in the future? 

Here's the thread from March.. https://github.com/unjs/nitro/discussions/1089. And re-reading that thread .. the feature is already available on `devHandlers`. Surely we can make it available on `handlers`!?

I've also posted about it in the nuxt repo https://github.com/nuxt/nuxt/discussions/24968.

**Comment by pi0** [comment]
You can register route handlers for production using `handlers` option. (`nitro.handlers` for nuxt).

You cannot add a function directly from your configuration. Rollup or any other bundler can't parse and bundle a javascript function this way.

Although it would be nice to hear more why you particularly need doing this I might then be able to suggest some other alternative solutions.
